### PR TITLE
Fix the ACL problem issue 4616

### PR DESCRIPTION
**Pull request by annando**
This fixes issue https://github.com/friendica/friendica/issues/4616

Additionally the line endings of several newly created files had been converted to the unix format.

### CODE REVIEW

**Comment by MrPetovan** [comment]
Can you point to the actual changes? In the current state, this PR is impossible to review because the whole ACL.php file is marked as modified in the commit that says "Fix ACL data".

Suggestion for next time: First commit all the line endings conversions, then commit the code changes separately.

**Comment by annando** [comment]
The only change had been with the ```ficxACL``` function that I reintroduced in that class.

File: src/Core/ACL.php [hunk 1]
```diff
@@ -1,358 +1,368 @@
...
+		$allow_cid = $matches[1];
+		preg_match_all($acl_regex, defaults($user, 'allow_gid', ''), $matches);
+		$allow_gid = $matches[1];
+		preg_match_all($acl_regex, defaults($user, 'deny_cid', ''), $matches);
+		$deny_cid = $matches[1];
+		preg_match_all($acl_regex, defaults($user, 'deny_gid', ''), $matches);
+		$deny_gid = $matches[1];
+
+		// Reformats the ACL data so that it is accepted by the JS frontend
+                array_walk($allow_cid, 'self::fixACL');
```
Comment (MrPetovan) [inline src/Core/ACL.php]: Standards: Please use tabs instead of spaces for indentation.

File: src/Core/ACL.php [hunk 2]
```diff
@@ -1,358 +1,368 @@
...
+                array_walk($allow_gid, 'self::fixACL');
+                array_walk($deny_cid, 'self::fixACL');
+                array_walk($deny_gid, 'self::fixACL');
```
Comment (MrPetovan) [inline src/Core/ACL.php]: Does this function reference notation works as well as `['self', 'fixACL']`?

File: src/Core/ACL.php (hunk 1 above)
Comment (annando) [reply src/Core/ACL.php]: typical cut&paste problem ;-)

File: src/Core/ACL.php (hunk 2 above)
Comment (annando) [reply src/Core/ACL.php]: I searched the web and found this as a solution.

Comment (MrPetovan) [reply src/Core/ACL.php]: Ok good!

Comment (annando) [reply src/Core/ACL.php]: See here: https://stackoverflow.com/questions/3840294/how-do-i-use-a-class-method-as-a-callback-function

**Comment by tobiasd** [comment]
> Line ending conversion makes reviewing unnecessarily complicated.

Yes. I think they should be in separate PRs if possible.

**Pull request merged**